src/ducc0/infra/threading.h

```diff
@@ -53,21 +53,92 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef DUCC0_THREADING_H
 #define DUCC0_THREADING_H
 
+// FIXME: This feels rather clumsy. Is there a better way to achieve this?
```

Comment:
`DUCC0_NO_THREADING` is gone; that is now covered by `DUCC0_LOWLEVEL_THREADING=DUCC0_NO_LOWLEVEL_THREADING`. So it exists as one of several options, not something that can be switched on and off independently.

As you say, a new choice of low-level threading would be added by, say,
```
#define DUCC0_TENSORFLOW_LOWLEVEL_THREADING 2
#define DUCC0_LOWLEVEL_THREADING DUCC0_TENSORFLOW_LOWLEVEL_THREADING
```
and adding the appropriate `#elif` sections in "threading.h" and "threading.cc"
(I' still unhappy about the explicit 0,1,2 count in the macros, but it'll have to do until I hear of something better.)
So adding a new threading mechanism is still a bit invasive, but should be much more localized than before.
I'd suggest not to use something generic like `CUSTOM_THREADING` but rather give it a descriptive name (containing `TENSORFLOW` or `GOOGLE` in your case). That way we can have more than one "nonstandard" extension.

Reply:
We have no plans to upstream other backends to `ducc` directly.  We would want to handle it internally.  To support this, we would need the ability to inject definitions of the mutex/lock/cv primitives into the `threading.h`.  LAPACK handles this kind of think with an optional `lapacke_config.h` header.  So it would be something like
```
// threading-specific headers
#if DUCC0_LOWLEVEL_THREADING == DUCC0_STDCXX_LOWLEVEL_THREADING
#include <mutex>
#include <condition_variable>
#elif DUCC0_LOWLEVEL_THREADING == DUCC0_NO_LOWLEVEL_THREADING
// no headers needed
#else
#include "ducc0_threading_config.h"  // Required for setting up custom threading primitives.
#endif
```

I don't see any value in supporting multiple non-standard extensions.  They could never coexist anyways.  As a user, either you disable threading entirely, use `ducc0`'s built-in cxx threading, or you're on your own.

Reply:
OK, I'll think about this; maybe we can use this to simplify the macro jungle a little. I hope to have an update in a few hours.